### PR TITLE
Consistent jar handling for Local and Remote UDFs

### LINKED ISSUE #848: Consistent jar handling for Local and Remote UDFs
The current build process for local UDFs generates a .function.json file in the build folder, but the JAR is not copied there and is instead used from the root folder. For remote UDFs, while the JARs are placed in the build folder, the jarPath in .function.json incorrectly points to the root folder, leading to inconsistent usage.

Objective:
- Ensure local UDF JARs are copied to the build folder.
- Ensure the jarPath in .function.json correctly references the JAR in the build folder for both local and remote UDFs.

### PR DESCRIPTION
**Pull request by nandorsoma**
This pull request introduces consistent handling of jar files for Local and Remote User-Defined Functions (UDFs) by updating the ResourceResolver interface and its implementations. This ensures paths are correctly resolved and utilized throughout the codebase.

Fixes #848

### CODE REVIEW

File: sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/preprocess/JarPreprocessor.java [hunk 1]
```diff
@@ -89,8 +89,24 @@ private Void processJarEntry(JarEntry entry, JarFile file,
       processorContext.addLibrary(path);
     }
 
+    copyFileWithStructure(path, processorContext.getBuildDir());
```
Comment (henneberger) [inline sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/preprocess/JarPreprocessor.java:92]: processorContext.addDependency(path); to copy files

File: sqrl-base/src/main/java/com/datasqrl/module/resolver/FileResourceResolver.java [hunk 2]
```diff
@@ -51,4 +47,13 @@ public Optional<Path> resolveFile(NamePath namePath) {
     }
     return Optional.of(path);
   }
+
+  @Override
+  public Optional<Path> resolve(Path path) {
```
Comment (henneberger) [inline sqrl-base/src/main/java/com/datasqrl/module/resolver/FileResourceResolver.java:52]: resolve(Path relativePath) ?

**Review by henneberger — CHANGES_REQUESTED** [review]
see comments

File: sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/preprocess/JarPreprocessor.java (hunk 1 above)
Comment (nandorsoma) [reply sqrl-tools/sqrl-packager/src/main/java/com/datasqrl/packager/preprocess/JarPreprocessor.java]: oh wow. cool!

**Review by henneberger — APPROVED** [review]

**Pull request merged**